src/pages/Facility/settings/devices/components/DeviceForm.tsx

```diff
@@ -63,10 +63,19 @@ const formSchema = z
         (date) => !date || isBefore(new Date(date), startOfTomorrow()),
         t("manufacture_date_cannot_be_in_future"),
       ),
+    registered_name: z.preprocess(
+      (value) => (typeof value === "string" ? value.trim() : value),
+      z
+        .string()
+        .min(1, { message: t("enter_atleast_one_character") })
+        .regex(/^[a-zA-Z\s-_]+$/, {
+          message: t("registered_name_must_contain_only_letters_spaces"),
+        }),
```

Comment:
I don't see any reason to restrict special characters. Doesn't harm allowing it.